frontend/src/metabase-lib/parameters/utils/filters.js

```diff
@@ -25,6 +26,26 @@ export function fieldFilterForParameter(parameter) {
   return () => false;
 }
 
+export function columnFilterForParameter(parameter) {
+  const type = getParameterType(parameter);
+  switch (type) {
+    case "date":
+      return column => Lib.isDate(column);
+    case "id":
+      return column => Lib.isPrimaryKey(column) || Lib.isForeignKey(column);
+    case "category":
+      return column => Lib.isCategory(column);
+    case "location":
+      return column => Lib.isLocation(column);
+    case "number":
+      return column => Lib.isNumber(column) && !Lib.isLocation(column);
```

Comment:
Do we need to check for `Lib.isLocation` if it's going to be checked before `case "number"` ?

If we do, then number could be a PK or FK as well.
Same goes for `case "string"`.

Reply:
Not sure. Current checks will do the same as the old ones.

Reply:
I'm pretty sure it does not matter in the body of this function.

>  Current checks will do the same as the old ones.

Previously `isCoordinate` was used for numbers, now `isLocation` is used.

Reply:
Yep, it's different but would result in the same output because of type hierarchy. E.g. Location can be Address or Coordinate, and only Coordinate are numbers.

Reply:
I did this because it would easier to follow switch logic where we previously check for `Location`